### PR TITLE
Test and display keybindings

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR follows on from #580 by adding the option to display all our keybindings in the pop-up from the help menu. It also adds more comprehensive tests for the keybindings themselves, including some tests similar to those in #544.

It also adjusts the stylings of both the `napari info` and `keybindings` pop-ups to use our style sheets.

The actual strings that are displayed for our keybindings clearly still need some work - particularly those associated with functions that don't have good doc strings or use a lambda

Finally I'm not quite so sure about the dialog box completely blocking the threads on the main viewer - say for keybindings it might be nice to have the keybindings box open and then be working with the viewer at the same time.

Here's what it looks like (note the actual menubar is cutoff from my gif!) - 
keybindings_display

Curious to get feedback from @AhmetCanSolak, @jni, and @royerloic at this stage though 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How has this been tested?

- [x] added a custom keybindings test and tests of all our keybindings in our viewer / layers.

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sofroniewn** [comment]
Note this will close #518

**Review by jni — APPROVED** [review]
Gorgeous! I love the nice use of numpydoc. =D

**Comment by jni** [comment]
Ah, just looked at the screenshot LOL. I reckon the lambdas vindicate my earlier advocacy for proper functions taking in an event=None even if they don't use the event info. =)

**Comment by sofroniewn** [comment]
@jni - many of those lambdas are actually about how to get the `Viewer` in as a keyword argument - for example 
```python
Viewer.bind_key('Control-E', lambda v: v.dims._roll())
```
not sure how to make this better.

We've definitely still got more to do here with our about menus / doc strings etc. but do you want to merge this PR now or are there more changes I should make before it goes in?

I'd probably say lets just merge and then we can keep going

**Comment by jni** [comment]
@sofroniewn 

```python
def roll_axes(viewer):
    """Change the order of the visible axes, e.g. [0, 1, 2] -> [2, 0, 1]."""
    viewer.dims._roll()


Viewer.bind_key('Control-E', roll_axes)
```

?

**Comment by sofroniewn** [comment]
that looks nicer! I can change all our lambdas over to that / or use the event trick so that we get nice doc strings for all of them. i'll do that before merging

**Comment by royerloic** [comment]
A Popup window...
But, ok, as long as this window is not modal (blocking the main window)
that should be fine...  ;-))
Actually super useful!

**Comment by jni** [comment]
@royerloic pop-ups are the standard way to do help menus. In the future, we can use vispy text to overlay the current shortcuts on the image when pressing '?'. Pressing any key would make the overlay disappear and activate the shortcut if the key was a shortcut. This is how Gmail keyboard shortcuts work and it's super handy.

**Comment by sofroniewn** [comment]
Right now the pop is blocking unfortunately @royerloic - we can work on fixing that in the future, will have to ensure that it remains up to date even when aspects of the viewer change for the keybindings window

**Comment by royerloic** [comment]
A floating -- non-blocking -- window that is visible and on the side, available to consult while working in napari, is not a bad thing I think... @sofroniewn indeed, syncing with potentially changing shortcuts could be done if we have some basic notification to that floating window that something change would be important in that case. I am happy with this as long as this is just a first prototype.... :-)

**Comment by sofroniewn** [comment]
@royerloic - i've made the window non-blocking, syncing so it updates when needed, and resizable - (though the resizing isn't quite perfect, it's pretty close)

I've also improved all the doc strings - I think it's pretty cool right now

I will merge after tests pass

**Pull request merged**
